app/scripts/migrations/147.ts

```diff
@@ -0,0 +1,72 @@
+import { AuthenticationControllerState } from '@metamask/profile-sync-controller/auth';
+import { hasProperty, isObject } from '@metamask/utils';
+import { cloneDeep } from 'lodash';
+
+type previousSessionDataShape = {
+  profile: {
+    identifierId: string;
+    profileId: string;
+  };
+  accessToken: string;
+  expiresIn: string;
+};
+
+export type VersionedData = {
+  meta: {
+    version: number;
+  };
+  data: {
+    AuthenticationController?: {
+      isSignedIn: boolean;
+      sessionData?:
+        | AuthenticationControllerState['sessionData']
+        | previousSessionDataShape;
+    };
+  };
+};
+
+export const version = 147;
+
+function transformState(state: VersionedData['data']) {
+  if (
+    !hasProperty(state, 'AuthenticationController') ||
+    !isObject(state.AuthenticationController)
+  ) {
+    global.sentry?.captureException?.(
+      new Error(
+        `Invalid AuthenticationController state: ${typeof state.AuthenticationController}`,
+      ),
+    );
+    return state;
+  }
+
+  // If the sessionData is a previous version, we need to delete it
+  if (
+    hasProperty(state.AuthenticationController, 'sessionData') &&
+    isObject(state.AuthenticationController.sessionData) &&
+    hasProperty(state.AuthenticationController.sessionData, 'accessToken')
+  ) {
+    state.AuthenticationController.sessionData = undefined;
```

Comment:
@mathieuartu this assumes that there will be a wallet unlock event after this migration is run, to re-login and rebuild the session data in the new shape, correct?